src/Tools/dotnet-monitor/CollectionRules/Actions/LoadProfilerAction.cs

```diff
@@ -0,0 +1,115 @@
+
+            return new LoadProfilerAction(_serviceProvider, endpointInfo, options);
+        }
+
+        internal sealed partial class LoadProfilerAction :
+            CollectionRuleActionBase<LoadProfilerOptions>
+        {
+            private readonly IEndpointInfo _endpointInfo;
+            private readonly LoadProfilerOptions _options;
+
+            public override CollectionRuleActionExecutionType ExecutionType
+            {
+                get => CollectionRuleActionExecutionType.AllowAsync;
+            }
+
+            public LoadProfilerAction(IServiceProvider serviceProvider, IEndpointInfo endpointInfo, LoadProfilerOptions options)
+                : base(endpointInfo, options)
+            {
+                _endpointInfo = endpointInfo ?? throw new ArgumentNullException(nameof(endpointInfo));
+                _options = options ?? throw new ArgumentNullException(nameof(options));
+
+                if (null == _options.Path || _options.PathMap.Count == 0)
+                {
+                    throw new ArgumentException(nameof(options.Path));
+                }
+
+                if (Guid.Empty == _options.ProfilerGuid)
+                {
+                    throw new ArgumentException(nameof(options.ProfilerGuid));
+                }
+            }
+
+            protected override async Task<CollectionRuleActionResult> ExecuteCoreAsync(
+                TaskCompletionSource<object> startCompletionSource,
+                CancellationToken token)
+            {
+                try
+                {
+                    DiagnosticsClient client = new DiagnosticsClient(_endpointInfo.Endpoint);
+
+                    ProcessInfo procInfo = await client.GetProcessInfoAsync(token);
+                    RidResolver ridResolver = new(procInfo);
+                    string choosenRid = ridResolver.Resolve(_options.PathMap.Keys);
+
+                    await client.SetStartupProfilerAsync(_options.ProfilerGuid, _options.PathMap[choosenRid], token);
+
+                    if (!startCompletionSource.TrySetResult(null))
+                    {
+                        throw new InvalidOperationException();
+                    }
+
+                    return new CollectionRuleActionResult()
+                    {
+                        OutputValues = new Dictionary<string, string>(StringComparer.Ordinal)
+                        {
+                            { CollectionRuleActionConstants.ProfilerLoadTypeValueName, Enum.Format(typeof(ProfilerLoadType), ProfilerLoadType.Startup, "G") },
+                            { CollectionRuleActionConstants.ProfilerSelectedRidValueName, choosenRid },
+                        }
+                    };
+                }
+                catch (Exception ex)
+                {
+                    throw new CollectionRuleActionException(ex);
+                }
+            }
+
+            internal enum ProfilerLoadType
+            {
+                Startup,
+                Attach,
```

Comment:
I don't see how we'll ever be able to deterministically support attach. There is no way to ask the runtime "are you in a state that allows profiler attach" without just doing it (same actually applies for startup, but we have better heuristic because it is extremely likely that the runtime is suspended when `dotnet monitor` is configured for listen mode). We would have to add some ability to poll a diagnostic command asking whether attach could ever occur for the given instance.